apiService.md

```diff
@@ -0,0 +1,77 @@
+# API Service
+
+## Prerequisites:
+
+- understanding of HTTP requests with [Fetch](https://developer.mozilla.org/en-US/docs/Web/API/fetch) and [Axios](https://www.npmjs.com/package/axios)
+- React [State](https://reactjs.org/docs/hooks-state.html) and [Context](https://reactjs.org/docs/context.html)
+
+## What is the API service?
+
+The API service is a way to organize HTTP requests logic needed to create, read, update and delete entries in data collection.
+
+## Getting started
+
+The following documentation is suited for developers with experience with the Common Ground methods of designing Progressive Web Applications and especially React Components from [Conduction](conduction.nl).
+
+## Previous situation
+
+In the previous situation, HTTP requests were made through the React built-in `fetch` methods, which required no imports in the components that used them, but using fetch equals a lot of redundant code (like using tokens as arguments on numerous occasions). Besides making the code less readable, it's also error-prone. Axios has a lot of advantages and if you want to read more about it, check out [this](https://betterprogramming.pub/why-javascript-developers-should-prefer-axios-over-fetch-294b28a96e2c) blog post. The main take here is you don't need a new instance in every `fetch` code block don't have to specify the URL and headers every time. This implementation will improve performance as well.
+
+### Key Differences
+
+- Axios needs to be imported
+
+```javascript
+    npm install axios
+```
+
+### apiServices
+
+Now, there is an `apiService.ts` file that takes care of all the headers in `src/apiService`. This file imports the `Resources` like Application, Attribute, Source, etc., from `/resources/`. It needs to import `axios , { AxiosInstance } from 'axios'`, along with the resources.
+
+It exports a class component with the tokens safely set privately and the `axiosClient` method to instantiate an HTTP request with `baseURL` set and Headers specified with tokens, content-type, and accepted returned formats.
+
+The apiServices call upon resources to instantiate
+
+### Resources
+
+The resources need instances (they get this from the apiService), this instance receives one of four methods (getAll, getOne, create, update). The resources are made separately for easy-to-read, maintainable, reusable components that allow custom calling. This way improves performance as if doesn't need to go through a maze of if-else statements.
```

Comment:
Top. Performance is debatable (heeft inderdaad effect, maar zal niet groot zijn). De instances krijgen niet per se één van de vier methoden. Ze hebben (meestal) vier voor CRUD functionaliteit (Create, Read, Update, Delete), maar kunnen ook andere methods hebben.

e.g. attribute krijgt het volgende ook mee:

```
public getAllFromEntity = (entityId: string): Promise<AxiosResponse> => {
    return this._instance.get(`/attributes?entity.id=${entityId}`)
  }
```

En logs (nog niet gebouwd) krijgt bijvoorbeeld geen delete. Dus hoe het omschreven staat klopt nu niet helemaal.

Verder wel spot-on.